package/kedro_viz/server.py

```diff
@@ -471,10 +472,74 @@ def format_pipeline_data(
             nodes[node_id]["pipelines"].append(pipeline_key)
 
 
+def _get_dataset_node(node_id, namespace):
+    if KEDRO_VERSION.match(">=0.16.0"):
+        try:
+            dataset = _CATALOG._get_dataset(namespace)
+        except DataSetNotFoundError:
+            dataset = None
+    else:
+        dataset = _CATALOG._data_sets.get(namespace)
+    _NODES[node_id] = {"type": "data", "obj": dataset}
+
+
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    node = _NODES.get(node_id)
+    if not node:
+        abort(404, description="Invalid node ID.")
```

Comment:
Following this for 404 error https://flask.palletsprojects.com/en/1.1.x/patterns/errorpages/#returning-api-errors-as-json

Reply:
Don't you need an `@app.errorhandler(404)` then? As `abort` [raises an `HTTPException`](https://flask.palletsprojects.com/en/1.1.x/api/#flask.abort).